src/app/core/services/sage300/sage300-configuration/sage300-connector.service.ts

```
@@ -32,9 +58,77 @@ export class Sage300ConnectorService {
   })
-  connectSage300(data: Sage300Credential): Observable<Sage300Credential> {
+  upsertCredentials(data: Sage300Credential): Observable<Sage300Credential> {
     globalCacheBusterNotifier.next();
+    if (this.doesSage300CredentialsExist){
+      return this.apiService.patch(`/workspaces/${this.storageService.get('workspaceId')}/credentials/sage300/`, data);
+    }
     return this.apiService.post(`/workspaces/${this.storageService.get('workspaceId')}/credentials/sage300/`, data);
   }
 
+  @Cacheable()
+  checkSage300TokenHealth(workspaceId: string): Observable<{}> {
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)